src/scales.js

```diff
@@ -105,3 +116,22 @@ function inferScaleType(key, channels, {type, domain, range}) {
 function asOrdinalType(key) {
   return registry.get(key) === position ? "point" : "ordinal";
 }
+
+// prepare scales for exposure through the plot's scales() function
+export function exposeScales(scaleDescriptors) {
+  return Object.fromEntries(
+    Object.entries(scaleDescriptors)
+    .map(([key, descriptor]) => [key, exposeScale(descriptor)])
+  );
+}
+
+function exposeScale({scale, label}) {
+  return {
+    domain: scale.domain(),
+    range: scale.range(),
+    ...scale.interpolate && {interpolate: scale.interpolate()},
+    ...label !== undefined && {label},
+    ...scale.type && {type: scale.type},
```

Comment:
```suggestion
    label,
    type: scale.type,
```

Can we always expose these? Seems fine for *label* to be undefined if missing, and I’m not sure when scale.type would be missing, if ever.